airflow/models/dag.py

```diff
@@ -2625,6 +2625,18 @@ def validate_schedule_and_params(self):
                     "DAG Schedule must be None, if there are any required params without default values"
                 )
 
+    @provide_session
+    def validate_task_pools(self, session=NEW_SESSION):
+        """Validates and raise exception if any task in a dag is using a non-existent pool"""
+        from airflow.models.pool import Pool
+
+        pools = [p.pool for p in Pool.get_pools(session)]
+        for task in self.tasks:
+            if task.pool not in pools:
+                raise PoolNotFound(
```

Comment:
In the event of multiple tasks with non-existent pools, is it preferable to fail as fast as possible and only warn about the first such occurrence, or should we first check all of the tasks and then raise with an error which reports all of the issues?

Reply:
I thought about that and feel that failing as fast as possible would be faster because a dag may have a lot of tasks and walking through all the tasks is likely to take time. WDYT?

Reply:
I think either way is fine, but looking into it more I think that most other DAG parser failure modes (ImportError, Syntax Errors, AirflowClusterPolicyViolation, etc) will fail on the first error, os we should probably do the same here.